WcaOnRails/spec/controllers/committee_positions_controller_spec.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe CommitteePositionsController, type: :controller do
+  let!(:committee) { FactoryGirl.create(:committee) }
+  let!(:committee_position) { FactoryGirl.create(:committee_position) }
+  let!(:committee_position_no_members) { FactoryGirl.create(:committee_position, name: "Committee position with no members") }
+  let!(:team_member) { FactoryGirl.create(:team_member) }
+
+  describe "GET #new" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :new, committee_id: committee_position.committee.slug
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create(:admin) }
+
+      it 'shows the committee positions new page' do
+        get :new, committee_id: committee_position.committee.slug
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'when signed in as a demoted admin' do
+      sign_in { FactoryGirl.create :admin_demoted }
+
+      it 'does not allow access' do
+        get :new, committee_id: committee_position.committee.slug
+        expect(response).to redirect_to root_url
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :new, committee_id: committee_position.committee.slug
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when not signed in' do
+      it 'redirects to the sign in page' do
+        post :create, committee_id: committee.slug, committee_position: {name: "An important position", description: "This is a very important position"}
```

Comment:
You can use `committee_position: FactoryGirl.attributes_for(:committee_position)`
Also it makes sense to use `let(:committee_position_attributes)` as you do this in multiple places.